app/main.py

```diff
@@ -16,13 +16,13 @@
 from .config import get_settings
 from .data import data_source
 from .routers import V1, V2
-from .utils.httputils import setup_client_session, teardown_client_session
+from .utils.httputils import Session
 
 # ############
 # FastAPI App
 # ############
 LOGGER = logging.getLogger("api")
-
+session = Session()
```

Comment:
What's the point of creating an instance of Session?

If there's only meant to be one Client Session for the lifetime of the application why are we tempting developers to create multiple Session instances by defining a class in the first place?

Seems everything could be a class method or attribute.
But at that point why have a class at all?